src/jvm/src/aws/sdk/kotlin/crt/auth/signing/AwsSignerJVM.kt

```diff
@@ -19,15 +19,35 @@ import software.amazon.awssdk.crt.auth.signing.AwsSigningConfig.AwsSignatureType
 import software.amazon.awssdk.crt.auth.signing.AwsSigningConfig.AwsSignedBodyHeaderType as AwsSignedBodyHeaderTypeJni
 import software.amazon.awssdk.crt.auth.signing.AwsSigningConfig.AwsSigningAlgorithm as AwsSigningAlgorithmJni
 
+/**
+ * Static class for a variety of AWS signing APIs.
+ */
 public actual object AwsSigner {
-    public actual suspend fun signRequest(request: HttpRequest, config: AwsSigningConfig): HttpRequest {
+
+    /**
+     * Signs an http request according to the supplied signing configuration
+     * @param request http request to sign
+     * @param config signing configuration
+     * @return signed request
+     */
+    public actual suspend fun signRequest(request: HttpRequest, config: AwsSigningConfig): HttpRequest =
+        checkNotNull(sign(request, config).signedRequest) { "AwsSigningResult request must not be null" }
```

Comment:
**Question**: What circumstances could lead to the signing result's request being null? Looking through the code I can't see how that would occur.

Reply:
Based on the comments from crt-java:
https://github.com/awslabs/aws-crt-java/blob/main/src/main/java/software/amazon/awssdk/crt/auth/signing/AwsSigningResult.java#L15-L19

I think it's due to the flexibility in the signing implementation but in practice it shouldn't be null the way we use it. Or at least that's my understanding.